### PR TITLE
user sign in sign up reset password

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- [#TicketID](https://edu-redmine.sun-asterisk.vn/issues/45394)

## WHAT (optional)
- User Sign in, Sign up, Reset password (send email)

## HOW
- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

## Evidence (Screenshot or Video)
Screenshot from 2022-01-25 15-26-52

image

### CODE REVIEW

**Review by hoangtv-2888 — APPROVED** [review]

**Comment by ghost** [comment]
ready

File: app/controllers/password_resets_controller.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      flash[:danger] = t "reset_failed"
+      render :edit
+    end
+  end
+
+  def edit; end
+
+  private
+
+  def get_user
```
Comment (huongnt-2545) [inline app/controllers/password_resets_controller.rb]: tránh đặt tên method có tiền tố get/set em nhé => `load_user`

File: app/mailers/application_mailer.rb [hunk 2]
```diff
@@ -1,2 +1,4 @@
 class ApplicationMailer < ActionMailer::Base
+  default from: "from@example.com"
```
Comment (huongnt-2545) [inline app/mailers/application_mailer.rb]: dùng biến env em nhé

**Comment by ghost** [comment]
ready

File: config/routes.rb [hunk 3]
```diff
@@ -1,4 +1,6 @@
 Rails.application.routes.draw do
+  get 'password_resets/new'
```
Comment (huongnt-2545) [inline config/routes.rb:2]: xóa 2 dòng này đi em, có khai báo ở dòng 16 rồi kìa

**Review by huongnt-2545 — APPROVED** [review]
merged

**Pull request merged**